src/domain/interfaces/prices-api.interface.ts

```diff
@@ -0,0 +1,18 @@
+import { AssetPrice } from '../prices/entities/asset-price.entity';
+
+export const IPricesApi = Symbol('IPricesApi');
+
+export interface IPricesApi {
+  getNativeCoinPrice(args: {
+    nativeCoinId: string;
```

Comment:
I would change `nativeCoinId` to be the `chainId` instead. Here we are "leaking" the implementation details on how Coingecko works to the rest of the application.

If Coingecko is not used as a price provider this function might not make sense anymore (in the new provider).

Reply:
I agree, good point. I'll move the `chainId-to-nativeCoinId` mapping (i.e. get the `nativeCoinId` from the app configuration) to the Coingecko-specific classes 👍